src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,35 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LENGTH_LOGIN_AND_PASSWORD = 6;
+    private static final int MIN_USER_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        isValidUser(user);
+        return storageDao.add(user);
+    }
+
+    private void isValidUser(User user) {
+        if (user == null) {
+            throw new RegistrationException("You must fill in all fields");
+        }
+        if (user.getId() == null) {
+            throw new RegistrationException("Id must be a number!");
+        }
+        if (user.getPassword() == null
+                || user.getPassword().length() < MIN_LENGTH_LOGIN_AND_PASSWORD) {
+            throw new RegistrationException("Password must be longer than 6 including");
+        }
+        if (user.getLogin() == null || user.getLogin().length() < MIN_LENGTH_LOGIN_AND_PASSWORD) {
+            throw new RegistrationException("Login must be longer than 6 including");
```

Comment:
please do not hardcode the values like `6` in the message. use values from constants. 
fix all similar places in the code please.